cogs/customreactions.py

```diff
@@ -0,0 +1,1111 @@
+                dm = False
+                cache_msg = await message.channel.fetch_message(message.id)
+                for reaction in cache_msg.reactions:
+                    users_who_reacted = await reaction.users().flatten()
+                    if author in users_who_reacted:
+                        if reaction.emoji == "1️⃣":
+                            delete = True
+                        if reaction.emoji == "2️⃣":
+                            anywhere = True
+                        if reaction.emoji == "3️⃣":
+                            dm = True
+
+                current_options.clear()
+                await message.clear_reactions()
+                conn = sqlite3.connect(self.bot.config.db_path)
+                c = conn.cursor()
+                t = (prompt_message, response, author.id, delete, anywhere, dm,
+                     not is_moderator)
+                c.execute(
+                    'INSERT INTO CustomReactions(Prompt, Response, UserID, DeletePrompt, Anywhere, DM, '
+                    'Proposal) VALUES(?,?,?,?,?,?,?)', t)
+                conn.commit()
+                conn.close()
+                self.rebuild_reaction_list()
+                self.rebuild_proposal_list()
+
+                if is_moderator:
+                    title = "Custom reaction successfully added!"
+                else:
+                    title = "Custom reaction proposal successfully submitted!"
+                description = "-Prompt: {}\n-Response: {}".format(
+                    prompt_message, response)
+                if delete:
+                    description = "{}\n-Will delete the message that calls the reaction".format(
+                        description)
+                if anywhere:
+                    description = "{}\n-Will activate the custom reaction if the prompt is anywhere in a message"\
+                        .format(description)
+                if dm:
+                    description = "{}\n-Will react in the DMs of the user who calls the reaction instead of the "\
+                                  "channel".format(description)
+
+                embed = discord.Embed(title=title, description=description)
+                embed.set_footer(text="Added by {}.".format(author),
+                                 icon_url=author.avatar_url)
+                await message.edit(embed=embed)
+
+                return
+
+            if reaction_emoji == "⏹":
+                await leave(message)
+                return
+
+        async def listcustomreacts(message, proposals):
+            if proposals:
+                current_list = self.proposal_list
+            else:
+                current_list = self.reaction_list
+
+            if not current_list:
+                if proposals:
+                    title = 'There are currently no custom reaction proposals in this server'
+                else:
+                    title = 'There are currently no custom reactions in this server'
```

Comment:
more duplicate string literals. instead, use (and with a better name than what I just came up with 😛 ):

```python
NO_CUSTOM = "There are currently no custom {} in this server"
NO_CUSTOM.format("reaction proposals")
NO_CUSTOM.format("reactions")
```

Reply:
See https://github.com/idoneam/Canary/pull/273#issuecomment-656869764